src/main/java/com/risevision/bigquery/BigqueryJobsApi.java

```diff
@@ -0,0 +1,63 @@
+package com.risevision.bigquery;
+
+import com.google.api.client.http.HttpTransport;
+import com.google.api.services.bigquery.model.*;
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.logging.Logger;
+
+public class BigqueryJobsApi extends BigqueryCommonApi implements JobsApi{
+
+  BigqueryJobsApi(HttpTransport transport) {
+    super(transport);
+  }
+
+  static final Logger log = Logger.getAnonymousLogger();
+
+  @Override
+  public void insertJob(JobInserterConfig configuration,
+                        JobInserterConfig.IndividualQueryConfiguration queryConfiguration) {
+    try {
+      Job job = assembleJob(configuration, queryConfiguration);
+      bqClient.jobs().insert(configuration.projectId, job).execute();
+    } catch (IOException e) {
+      log.severe("Could not insert job " + e.getMessage());
+    }
+  }
+
+  protected Job assembleJob(JobInserterConfig configuration,
+                  JobInserterConfig.IndividualQueryConfiguration queryConfiguration) throws IOException {
+
+    Job job = new Job();
+    JobConfiguration jobConfiguration = new JobConfiguration();
+    JobConfigurationQuery jobConfigurationQuery = new JobConfigurationQuery();
+    JobReference jobReference = new JobReference();
+    String sqlQuery = readResource(queryConfiguration.fileName);
+    TableReference destinationTable = new TableReference();
+
+    // Prepare BigQuery request objects
+    job.setJobReference(jobReference);
+    job.setConfiguration(jobConfiguration);
+    jobConfiguration.setQuery(jobConfigurationQuery);
+    jobConfigurationQuery.setDestinationTable(destinationTable);
+
+    // Configuration common to all jobs
+    jobReference.setProjectId(configuration.projectId);
+    jobConfigurationQuery.setWriteDisposition(configuration.configuration.query.writeDisposition);
+    destinationTable.setProjectId(configuration.configuration.query.destinationTable.projectId);
+    destinationTable.setDatasetId(configuration.configuration.query.destinationTable.datasetId);
+
+    // Job-specific configuration
+    jobConfigurationQuery.setQuery(sqlQuery);
+    destinationTable.setTableId(queryConfiguration.destinationTableId);
+
+    return job;
+  }
+
+  String readResource(String fileName) throws IOException {
```

Comment:
I think it's reasonable to crash on config errors unless we want to add some kind of alerting mechanism.  But I prefer having all the config loaded statically at startup so that the crash is immediate.  I was going to ask why you have multiple configuration files for the job insert task rather than just including the sql as part of the rest of the configuration.  I didn't bother because there was already enough on this PR and there does seem to be some additional cleanliness in having sql separated.  So I think the only suggestion I would have here is to move the sql queries into the other config file and read config only once at startup.  But not really worth changing things over.